creator/studies/migrations/0029_add_populate_domain.py

```diff
@@ -0,0 +1,63 @@
+# Generated by Django 2.2.24 on 2021-11-02 18:23
+
+"""
+This migration creates the domain attribute for the Study model and also
+populates it using the spreadsheet found in this migrations folder.
+"""
+
+from django.db import migrations, models
+
+import pandas as pd
+import os
+
+
+def set_domain(apps, schema_editor):
+    """
+    Populate domain for all Study objects using the provided data.
+    """
+    df = pd.read_csv(
+        os.path.abspath("creator/studies/migrations/study_metadata.csv")
+    )
+    domains = pd.Series(df.domain.values, index=df.kf_id).to_dict()
+    Study = apps.get_model("studies", "Study") 
+    mappings = {
+        "Birth Defect": "BIRTHDEFECT",
+        "Other": "OTHER",
+        "Cancer": "CANCER",
+        "Cancer,Birth Defect": "CANCERBIRTHDEFECT",
+        "COVID-19": "COVID19",
+    }
+
+    for study in Study.objects.all():
+        kf_id = study.kf_id
+        if kf_id in domains:
+            study.domain = mappings[domains[kf_id].strip()]
+        else:
+            study.domain = "UNKNOWN"
+        study.save()
+
+
+def undo_set_domain(apps, schema_editor):
+    """
+    Set domain to UNKNOWN for all Study objects.
+    """
+    Study = apps.get_model("studies", "Study")
+    for study in Study.objects.all():
+        study.short_code = None
```

Comment:
```suggestion
        study.domain = None
```

Reply:
Sorry I think I put you on the wrong path with my previous comments on the undo functions. When you rollback these migrations and it calls the undo functions, the rollbacks fail bc domain and program are non-nullable. I *think* that means you were right before when you were setting the values to unknown. And I believe we want the same thing for short_code.